src/main/java/io/strimzi/kafka/quotas/PolicyBasedThrottle.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020, Red Hat Inc.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.quotas;
+
+import java.util.Collection;
+import java.util.Objects;
+
+
+/**
+ * Converts volume usage to a throttle factor by applying a {@link ThrottleFactorPolicy} to the observations.
+ * Notifies a listener when the factor changes.
+ */
+public class PolicyBasedThrottle implements VolumeObserver, ThrottleFactorSource {
+
+    private final ThrottleFactorPolicy factorPolicy;
+    private final Runnable listener;
+    private volatile Double throttleFactor = 1.0d;
```

Comment:
This can be a primitive

Reply:
resolved with ed052f6ffe7180ae07791c5dcf4dff76dd8007ab